### PR TITLE
NDRS-691: Add instrumentation around ContractRuntime

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-691

Sample Output:
Screenshot from 2020-12-22 08-48-37

### CODE REVIEW

**Comment by goral09** [comment]
I would say we will also need timers/gauges for this. cc @EdHastingsCasperLabs

**Comment by darthsiroftardis** [comment]
> I would say we will also need timers/gauges for this. cc @EdHastingsCasperLabs

I thought we simply want to track the amount of times these two specific requests are made yes? there is already some instrumentation around the contract runtime

**Comment by goral09** [comment]
> > I would say we will also need timers/gauges for this. cc @EdHastingsCasperLabs
> 
> I thought we simply want to track the amount of times these two specific requests are made yes?

I think we also want to measure how long it takes. cc @EdHastingsCasperLabs @henrytill 

> there is already some instrumentation around the contract runtime

Do we measure timings of these endpoints though?

**Comment by henrytill** [comment]
I don't remember the exact context of this work and the ticket isn't much help, so it's hard for me to comment on the specifics here.  But in general I tend to agree with @goral09, timers are useful.

**Comment by darthsiroftardis** [comment]
@goral09 checking the code, we do track the timing of both requests, however they seem to be under the `get_balance` metric. I think it should be under a separate histogram metric for each request. I shall make that change

**Comment by darthsiroftardis** [comment]
@goral09 I think the PR is sufficient now

File: node/src/components/contract_runtime.rs [hunk 1]
```diff
@@ -101,6 +103,12 @@ const GET_BALANCE_NAME: &str = "contract_runtime_get_balance";
 const GET_BALANCE_HELP: &str = "tracking run of engine_state.get_balance.";
 const GET_VALIDATOR_WEIGHTS_NAME: &str = "contract_runtime_get_validator_weights";
 const GET_VALIDATOR_WEIGHTS_HELP: &str = "tracking run of engine_state.get_validator_weights.";
+const GET_ERA_VALIDATORS_TIMING_NAME: &str = "contract_runtime_get_era_validators";
```
Comment (goral09) [inline node/src/components/contract_runtime.rs:106]: the `_TIMING_` part is unnecessary.

File: node/src/components/contract_runtime.rs [hunk 2]
```diff
@@ -78,6 +78,10 @@ pub struct ContractRuntimeMetrics {
     run_query: Histogram,
     get_balance: Histogram,
     get_validator_weights: Histogram,
+    get_era_validators_timing: Histogram,
+    get_era_validator_weights_by_era_id_timing: Histogram,
+    get_era_validators: Counter,
+    get_era_validator_weights_by_era_id: Counter,
```
Comment (goral09) [inline node/src/components/contract_runtime.rs:84]: We don't need counters if we have the histogram.

File: node/src/components/contract_runtime.rs (hunk 1 above)
Comment (darthsiroftardis) [reply node/src/components/contract_runtime.rs]: Removed `_TIMING_`

File: node/src/components/contract_runtime.rs (hunk 2 above)
Comment (darthsiroftardis) [reply node/src/components/contract_runtime.rs]: Removed counters

**Review by goral09 — APPROVED** [review]

**Pull request merged**